example/constructor-stats.h

```diff
@@ -0,0 +1,176 @@
+#pragma once
```

Comment:
Might be good to add just a minor comment like:

This class provides helper functions for tracking and printing information about the lifetime of instances created by pybind11 to ensure that future changes do not introduce reference leaks, etc. etc.